security/src/main/java/io/airlift/security/client/JettyAuthenticationStore.java

```diff
@@ -0,0 +1,125 @@
+package io.airlift.security.client;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.eclipse.jetty.client.api.Authentication;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class JettyAuthenticationStore implements AuthenticationStore
```

Comment:
This is not a very good name. Jetty already has an authentication store. A more descriptive name would be better (e.g., `ExpiringAuthenticationStore`, or something like that)